# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: js/src/jit/Recover.cpp
# Commit: 22854b710fbf
# Full Hash: 22854b710fbfb779ac6b19fe4c9ac2570189cec0
# Author: Jan de Mooij <jdemooij@mozilla.com>
# Date: 2021-03-06 09:33:16
# Regressor Bug: 1689413
# File Overlap Count: 1
# Description:
#   Bug 1689413 part 6 - Move Realm* from ObjectGroup to BaseShape. r=jonco
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D106976
# ==============================================================================

diff -r 4531d5c25694 -r 22854b710fbf js/src/jit/Recover.cpp
--- a/js/src/jit/Recover.cpp	Fri Mar 05 19:10:06 2021 +0000
+++ b/js/src/jit/Recover.cpp	Fri Mar 05 19:10:07 2021 +0000
@@ -1571,9 +1571,9 @@
 bool RNewArray::recover(JSContext* cx, SnapshotIterator& iter) const {
   RootedObject templateObject(cx, &iter.read().toObject());
   RootedValue result(cx);
-  RootedObjectGroup group(cx, templateObject->group());
+  RootedShape shape(cx, templateObject->shape());
 
-  ArrayObject* resultObject = NewArrayWithGroup(cx, count_, group);
+  ArrayObject* resultObject = NewArrayWithShape(cx, count_, shape);
   if (!resultObject) {
     return false;
   }